pkg/apis/mysql/v1alpha1/mysqlcluster_status.go

```diff
@@ -0,0 +1,215 @@
+func (b *MysqlBackup) GetCondition(ty BackupConditionType) *BackupCondition {
+	for _, cond := range b.Status.Conditions {
+		if cond.Type == ty {
+			return &cond
+		}
+	}
+
+	return nil
+}
+
+// UpdateStatusCondition sets the condition to a status.
+// for example Ready condition to True, or False
+func (b *MysqlBackup) UpdateStatusCondition(condType BackupConditionType,
+	status core.ConditionStatus, reason, msg string) {
+	newCondition := BackupCondition{
+		Type:    condType,
+		Status:  status,
+		Reason:  reason,
+		Message: msg,
+	}
+
+	t := time.Now()
+
+	if len(b.Status.Conditions) == 0 {
+		glog.V(4).Infof("Setting lastTransitionTime for mysql backup "+
+			"%q condition %q to %v", b.Name, condType, t)
+		newCondition.LastTransitionTime = metav1.NewTime(t)
+		b.Status.Conditions = []BackupCondition{newCondition}
+	} else {
+		if i, exist := b.condExists(condType); exist {
+			cond := b.Status.Conditions[i]
+			if cond.Status != newCondition.Status {
+				glog.V(4).Infof("Found status change for mysql backup "+
+					"%q condition %q: %q -> %q; setting lastTransitionTime to %v",
+					b.Name, condType, cond.Status, status, t)
+				newCondition.LastTransitionTime = metav1.NewTime(t)
+			} else {
+				newCondition.LastTransitionTime = cond.LastTransitionTime
+			}
+			glog.V(4).Infof("Setting lastTransitionTime for mysql backup "+
+				"%q condition %q to %q", b.Name, condType, status)
+			b.Status.Conditions[i] = newCondition
+		} else {
+			glog.V(4).Infof("Setting new condition for mysql backup %q, condition %q to %q",
+				b.Name, condType, status)
+			newCondition.LastTransitionTime = metav1.NewTime(t)
+			b.Status.Conditions = append(b.Status.Conditions, newCondition)
+		}
+	}
+}
+
+func (b *MysqlBackup) condExists(ty BackupConditionType) (int, bool) {
+	for i, cond := range b.Status.Conditions {
+		if cond.Type == ty {
+			return i, true
+		}
+	}
+
+	return 0, false
+}
+
+// UpdateNodeCondition updates the condition for a given type
+func (ns *NodeStatus) UpdateNodeCondition(cType NodeConditionType,
+	cStatus core.ConditionStatus) bool {
+
+	newCondition := NodeCondition{
+		Type:   cType,
+		Status: cStatus,
+	}
+
+	changed := false
+	t := time.Now()
+
+	if len(ns.Conditions) == 0 {
+		glog.V(4).Infof("Setting lastTransitionTime for node "+
+			"%q condition %q to %v", ns.Name, cType, t)
+		newCondition.LastTransitionTime = metav1.NewTime(t)
+		ns.Conditions = []NodeCondition{newCondition}
+		changed = true
+	} else {
+		if i, exist := ns.condExists(cType); exist {
+			cond := ns.Conditions[i]
+			if cond.Status != newCondition.Status {
+				glog.V(4).Infof("Found status change for node "+
+					"%q condition %q: %q -> %q; setting lastTransitionTime to %v",
+					ns.Name, cType, cond.Status, cStatus, t)
+				newCondition.LastTransitionTime = metav1.NewTime(t)
+				changed = true
+			} else {
+				newCondition.LastTransitionTime = cond.LastTransitionTime
+			}
+			glog.V(4).Infof("Setting lastTransitionTime for node "+
```

Comment:
Remove glog from `pkg/apis` so that this package can be safely imported and used into other projects.